components/tidb_query_datatype/src/expr/ctx.rs

```diff
@@ -154,7 +154,8 @@ impl EvalConfig {
         EvalWarnings::new(self.max_warning_cnt)
     }
 
-    #[cfg(test)]
+    // TODO: builtin_cast test case can not cimpile if #[cfg(test)]
+    //    #[cfg(test)]
```

Comment:
why?

Reply:
Because tidb_query was splited, test utils marked with#[cfg(test)] in top crate can no longer be propagated to leaf crates. I try to use #[cfg(feature="testing")] but still can not compile. At last I rm #[cfg(test)] and rename those functions as xxx_for_test.